adapters/huaweiads/huaweiads.go

```diff
@@ -0,0 +1,1320 @@
+
+	var numImage = 0
+	var numVideo = 0
+	var width int64
+	var height int64
+	for _, asset := range nativePayload.Assets {
+		if asset.Video != nil {
+			numVideo++
+		}
+		// every image has the same W, H.
+		if asset.Img != nil {
+			numImage++
+			if asset.Img.H != 0 && asset.Img.W != 0 {
+				width = asset.Img.W
+				height = asset.Img.H
+			} else if asset.Img.WMin != 0 && asset.Img.HMin != 0 {
+				width = asset.Img.WMin
+				height = asset.Img.HMin
+			}
+		}
+	}
+	adslot30.W = width
+	adslot30.H = height
+
+	var detailedCreativeTypeList = make([]string, 0, 2)
+	if numVideo >= 1 {
+		detailedCreativeTypeList = append(detailedCreativeTypeList, "903")
+	} else if numImage > 1 {
+		detailedCreativeTypeList = append(detailedCreativeTypeList, "904")
+	} else if numImage == 1 {
+		detailedCreativeTypeList = append(detailedCreativeTypeList, "909")
+	} else {
+		detailedCreativeTypeList = append(detailedCreativeTypeList, "913", "914")
+	}
+	adslot30.DetailedCreativeTypeList = detailedCreativeTypeList
+	return nil
+}
+
+// convert adtype String -> Integer
+func convertAdtypeString2Integer(adtypeLower string) int32 {
+	if adtypeLower == "banner" {
+		return banner
+	} else if adtypeLower == "native" {
+		return native
+	} else if adtypeLower == "rewarded" {
+		return rewarded
+	} else if adtypeLower == "splash" {
+		return splash
+	} else if adtypeLower == "interstitial" {
+		return interstitial
+	} else if adtypeLower == "roll" {
+		return roll
+	} else {
+		return banner
+	}
+}
+
+// get app information for HuaweiAds request
+func getHuaweiAdsReqAppInfo(request *huaweiAdsRequest, openRTBRequest *openrtb2.BidRequest) error {
+	var app app
+	if openRTBRequest.App != nil {
+		if openRTBRequest.App.Ver != "" {
+			app.Version = openRTBRequest.App.Ver
+		}
+		if openRTBRequest.App.Name != "" {
+			app.Name = openRTBRequest.App.Name
+		}
+
+		// bundle cannot be empty, we need package name.
+		if openRTBRequest.App.Bundle != "" {
+			app.Pkgname = openRTBRequest.App.Bundle
+		} else {
+			return errors.New("HuaweiAdsReqApp: Pkgname is empty.")
+		}
+
+		if openRTBRequest.App.Content != nil && openRTBRequest.App.Content.Language != "" {
+			app.Lang = openRTBRequest.App.Content.Language
+		} else {
+			app.Lang = "en"
+		}
+	}
+	app.Country = getCountryCode(openRTBRequest)
+	request.App = app
+	return nil
+}
+
+// get field clientTime, format: 2006-01-02 15:04:05.000+0200
+func getClientTime(clientTime string) (newClientTime string) {
+	var zone = DefaultTimeZone
+	t := time.Now().Local().Format(time.RFC822Z)
+	index := strings.IndexAny(t, "-+")
+	if index > 0 && len(t)-index == 5 {
+		zone = t[index:]
+	}
+	if clientTime == "" {
+		return time.Now().Format(TimeFormat) + zone
+	}
+	if isMatched, _ := regexp.MatchString("^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}.\\d{3}[+-]{1}\\d{4}$", clientTime); isMatched {
+		return clientTime
+	}
+	if isMatched, _ := regexp.MatchString("^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}.\\d{3}$", clientTime); isMatched {
```

Comment:
Please add a json test case for this second regex.

Reply:
sorry，the zone contained in the result is dynamic, how to add a fixed value to verify it?